### PR TITLE
Issue #7583: Added Code example for default config of NeedBraces and other tokens

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Issue #7583: Added Code example for default config of NeedBraces and other tokens

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 1]
```diff
@@ -112,6 +138,30 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to not allow {@code case, default} single-line statements without braces:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NeedBraces&quot;&gt;
+ *  &lt;property name=&quot;tokens&quot; value=&quot;LITERAL_CASE, LITERAL_DEFAULT&quot;/&gt;
```
Comment (Vyom-Yadav) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:145]: Do I need to write `&lt;property name=&quot;allowSingleLineStatement&quot; value=&quot;false&quot;/&gt;`

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:145]: as mentioned here "[NeedBraces](https://checkstyle.org/config_blocks.html#NeedBraces)" default value of allowSingleLineStatement is false and you also asking for the same but i think you should write the property name if you have mentioned property name in below code example which you did.

Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:145]: The property name cannot be mentioned without giving it a value, and that what I am saying above.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:145]: property name with value okay but i thought u wanna know whether to write property name with default value.

Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:145]: ok np

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 2]
```diff
@@ -66,6 +66,32 @@
  * <pre>
  * &lt;module name="NeedBraces"/&gt;
  * </pre>
+ * <p>Valid code example:</p>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:69]: Just `Example:`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 3]
```diff
@@ -66,6 +66,32 @@
...
+ *    return false;
+ * }
+ * for(int i =0; ; ) { // OK
+ *    ++count;
+ * }
+ * do { // OK
+ *    ++count;
+ * } while (false);
+ * </pre>
+ * <p>Invalid code example:</p>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:83]: No need for separate section, add OK and not OK examples in one section

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 4]
```diff
@@ -66,6 +66,32 @@
...
  * &lt;module name="NeedBraces"/&gt;
  * </pre>
+ * <p>Valid code example:</p>
+ * <pre>
+ * if (true) { // OK
+ *    return true;
+ * }  else { //OK
+ *    return false;
+ * }
+ * for(int i =0; ; ) { // OK
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:76]: please format code correctly and make it more look like real life code (`(int i = 0; i < 5; i++)`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 5]
```diff
@@ -66,6 +66,32 @@
  * <pre>
  * &lt;module name="NeedBraces"/&gt;
  * </pre>
+ * <p>Valid code example:</p>
+ * <pre>
+ * if (true) { // OK
+ *    return true;
+ * }  else { //OK
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:73]: whitespace after `//`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 6]
```diff
@@ -112,6 +138,30 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to not allow {@code case, default} single-line statements without braces:
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:141]: I do not see benefit from this example. There is already several examples of config with tokens property overridden.
Please remove it

**Review by strkkk — CHANGES_REQUESTED** [review]

**Comment by Vyom-Yadav** [comment]
Github, rebase

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 6 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 5 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 4 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 3 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 4 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: Not done, code is not formatted

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 7]
```diff
@@ -66,6 +66,19 @@
  * <pre>
  * &lt;module name="NeedBraces"/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * if (true) { // OK
+ *    return true;
+ * }  else  // violation, single line statements without braces are not allowed
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:73]: `}  else` double space

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 8]
```diff
@@ -66,6 +66,19 @@
...
+ *    return false;
+ * for(int i =0; i &lt; 5 ; i++ ) { // OK
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:75]: `for(` space needed

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 9]
```diff
@@ -66,6 +66,19 @@
...
+ *    ++count;
+ * }
+ * do // violation, single line statements without braces are not allowed
+ *    ++count;
+ *  while (false);
+ * </pre>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:81]: please add violations for for- and while-cycles

**Review by strkkk — CHANGES_REQUESTED** [review]
Please also see comment above

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 9 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:85]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 8 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 7 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 4 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: I formatted it with 4 spaces.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 9 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:85]: Just a suggestion, I think we should add different code examples for valid and invalid as the code would look neater, rest up to you.

**Review by strkkk — APPROVED** [review]
Cirrus is failing for some infra issue

**Review by pbludov — APPROVED** [review]
I'm ok to merge.

**Comment by romani** [comment]
Github, generate website

**Comment by romani** [comment]
@Vyom-Yadav , 
please update all example blocks to reuse the same example java code, to let user see how changing of configuration make a difference in behavior on the same java code.
there are configurations for `To configure the check for if and else blocks: ` and ` To configure the check to allow single-line statements (if, while, do-while, for) without braces: ` which requires code example aslo - please do.

please share CLI output for each of your config+code from documentation to make sure that violations from CLI a matching violations in documentation.
example - https://github.com/checkstyle/checkstyle/pull/6103#issuecomment-417940670

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 10]
```diff
@@ -66,6 +66,23 @@
...
+ * } else // violation, single-line statements not allowed without braces
+ *     return false;
+ * for (int i = 0; i &lt; 5; i++) { // OK
+ *     ++count;
+ * }
+ * do // violation, single-line statements not allowed without braces
+ *     ++count;
+ * while (false);
+ * for (int j = 0; j &lt; 10; j++) // violation, single-line statements not allowed without braces
+ *                                 // violation, empty loop body not allowed
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:82]: > violation, empty loop body not allowed

please show by CLI that this violation is from Needbrace Check.

**Review by romani — CHANGES_REQUESTED** [review]
see comment above.

item:

**Review by rnveach — APPROVED** [review]

**Comment by Vyom-Yadav** [comment]
Using Checkstyle 8.38, with the config:
````
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http:///www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="NeedBraces">
            <property name="allowSingleLineStatement" value="FALSE"/>
            <property name="tokens" value="LITERAL_IF,LITERAL_ELSE"/>
        </module>
    </module>
</module>
````
The Code
````
public class Test {
    int x = 5;

    public final boolean check() {
        if (x > 10) {
            return true;
        } else
            return false;
    }
}
````
Terminal Output
```
Starting audit...
[ERROR] D:\workspace\configCheckstyle\src\cC\Test.java:9:11: 'else' construct must use '{}'s. [NeedBraces]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by Vyom-Yadav** [comment]
> @Vyom-Yadav ,
> please update all example blocks to reuse the same example java code, to let user see how changing of configuration make a difference in behavior on the same java code.
> there are configurations for `To configure the check for if and else blocks: ` and `To configure the check to allow single-line statements (if, while, do-while, for) without braces:` which requires code example aslo - please do.
> 
> please share CLI output for each of your config+code from documentation to make sure that violations from CLI a matching violations in documentation.
> example - [#6103 (comment)](https://github.com/checkstyle/checkstyle/pull/6103#issuecomment-417940670)

Already there is already an example for `To configure the check to allow single-line statements (if, while, do-while, for) without braces:`

**Comment by Vyom-Yadav** [comment]
Github, rebase

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java (hunk 10 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java:82]: It didn't show any error for empty loop body, opened an issue for same.

**Comment by Vyom-Yadav** [comment]
using configuration
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http:///www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="NeedBraces">
            <property name="allowSingleLineStatement" value="FALSE"/>
            <property name="allowEmptyLoopBody" value="FALSE"/>
            <property name="tokens" value="LITERAL_IF,LITERAL_ELSE,LITERAL_DO,LITERAL_WHILE,LITERAL_FOR"/>
        </module>
    </module>
</module>
```
Code example used-
```
public class Test {
    public final boolean check() {
        if (true) {
            return true;
        } else
            return false;
    }

    public static void main(String[] args) {
        int count = 0;
        int counter = 11;
        for (int i = 0; i < 5; i++) {
            ++count;
        }
        do
            ++count;
        while (false);
        for (int j = 0; j < 10; j++) ;
        while (counter < 10)
            ++count;
    }
}
```
Violations shown
```
Starting audit...
[ERROR] D:\workspace\configCheckstyle\src\cC\Test.java:7:11: 'else' construct must use '{}'s. [NeedBraces]
[ERROR] D:\workspace\configCheckstyle\src\cC\Test.java:17:9: 'do' construct must use '{}'s. [NeedBraces]
[ERROR] D:\workspace\configCheckstyle\src\cC\Test.java:20:9: 'for' construct must use '{}'s. [NeedBraces]
[ERROR] D:\workspace\configCheckstyle\src\cC\Test.java:21:9: 'while' construct must use '{}'s. [NeedBraces]
Audit done.
Checkstyle ends with 4 errors.
```
Not showing error for empty loop body checked it independently too.

**Comment by Vyom-Yadav** [comment]
Github. generate web site

**Comment by Vyom-Yadav** [comment]
Github, rebase

**Comment by Vyom-Yadav** [comment]
@romani this is completely done, kindly merge.

**Comment by Vyom-Yadav** [comment]
Tried Github Desktop, but it didn't work for me

**Pull request closed without merging**